tests/unit/test_helpers.py

```diff
@@ -499,6 +500,48 @@ def test_ab_init_active(self):
             ab_init(response)
             response.set_cookie.assert_not_called()
 
+    def test_get_extra_labels(self):
+        """Test extra_labels extraction from form object."""
+        with webcompat.app.test_request_context('/issues/new', method='POST'):
+
+            webcompat.app.preprocess_request()
+
+            self.assertEqual(get_extra_labels(
+                {'extra_labels': '["type-marfeel", "browser-fenix"]'}),
+                ['type-marfeel', 'browser-fenix']
+            )
+            self.assertEqual(get_extra_labels({'extra_labels': '[]'}), [])
+            self.assertEqual(get_extra_labels({}), None)
+            self.assertEqual(get_extra_labels({'extra_labels': ''}), None)
+
+    def test_get_extra_labels_for_experiment(self):
+        """Test extra_labels extraction from form object if
+        experiment is active."""
+        with webcompat.app.test_request_context(
+                '/',
+                method='POST'):
+
+            webcompat.app.config['AB_EXPERIMENTS'] = {
+                'exp': {
+                    'variations': {
+                        'form-v2': (0, 100)
+                    },
+                    'max-age': 86400
+                }
+            }
+            webcompat.app.preprocess_request()
```

Comment:
Maybe let's add a comment above this, that we need `webcompat.app.preprocess_request()` because the `g.current_experiments = ab_current_experiments()` is defined in `before_request`, which is executed in tests only if `webcompat.app.preprocess_request()` is explicitly called.

Reply:
thanks, added the comment